### ISSUE TITLE
Target module QuantLinear() is not supported. Currently, only torch.nn.Linear and Conv1D are supported

### ISSUE THREAD

**Issue by chienhalan5000** [oobabooga/text-generation-webui#2297]
### Describe the bug

I'm using "training tab" to train LoRa using gpt4-alpaca-lora-30B-GPTQ-4bit-128g model and got the error. The memories are allocated fine and not reached to OOM problem. I tested using gpt4-alpaca-lora-13B-GPTQ-4bit-128g model to train LoRA in the same process and everything just fine.

I also read the issue [#2183](https://github.com/oobabooga/text-generation-webui/issues/2183) but seem like not my case.

Please help me. Thanks.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

None

### Screenshot

![2023-05-19_17-46-49](https://github.com/oobabooga/text-generation-webui/assets/34291365/1bd5952e-0f30-4cc6-9094-5644c8014bbf)

### Logs

```shell
None
```

### System Info

```shell
Dual RTX 3060 (24GB VRAM in total). 32GB RAM
```

**Comment by Ph0rk0z** [comment]
Try different version of PEFT.

**Comment by thusinh1969** [comment]
This issue is still there ... :( ... Can someone help ?!?!?

Thanks
Steve

**Comment by Extraltodeus** [comment]
Same here!

**Comment by Ph0rk0z** [comment]
You have to load it with "monkeypatch" for this repo. I just saw this error when I loaded with regular GPTQ by mistake.

**Comment by chienhalan5000** [comment]
@Ph0rk0z When I load it with "monkeypatch" (using --monkeypatch), I got the OOM error. So first load without monkeypatch and then I turn on monkeypatch using the "Interface mode" tab and reload the model to apply the "monkeypatch" and it's just fine. I tested in the same process for the gpt4-alpaca-lora-13B-GPTQ-4bit-128g model and I'm able to fine-tune it.

Btw, I'm currently fresh install alpaca-lora-4bit repo and try using finetune script. Hoping for better luck :)

**Comment by Ph0rk0z** [comment]
With the 30b, the monkeypatch does use more memory 19 vs 17.5 gb. Not sure why, probably something in the kernel.

If you are truly low on memory enable gradient check pointing in that repo. Just don't over-train since it disables dropout.

**Comment by chienhalan5000** [comment]
After install alpaca_lora_4bit, I'm able to finetune the 30B model with gradient check pointing enabled, and I think this is the main point to solve the problem. Thanks!

**Comment by balaji-skoruz** [comment]
> After install alpaca_lora_4bit, I'm able to finetune the 30B model with gradient check pointing enabled, and I think this is the main point to solve the problem. Thanks!

can you guide me? -getting same error

**Comment by lucas0** [comment]
Also facing the same issue.. ):

**Comment by Ph0rk0z** [comment]
It needs to be installed and *compiled*. That's probably a tiny bit harder on windows. For gptq-for-llama I think there are pre compiled binaries shared.

**Comment by ArulinTheUnicorn** [comment]
I am seeing the exact same issue and I have the monkeypatch, please stop using the monkeypatch as your an escape monkey

**Comment by Ph0rk0z** [comment]
>Target module Autograd4bitQuantLinear() is not supported.

Right that means PEFT patch is not applied successfully. And PEFT has been changed a lot since this came out. The version in requirements.txt is probably no longer correct. Plus this is all windows users installing pre-compiled wheels.

This is the patch that has to succeed on PEFT: https://github.com/johnsmith0031/alpaca_lora_4bit/blob/main/monkeypatch/peft_tuners_lora_monkey_patch.py

I have not checked recently what else is going on as it's a good idea to used dedicated training tools to train.

**Comment by ArulinTheUnicorn** [comment]
> > Target module Autograd4bitQuantLinear() is not supported.
> 
> Right that means PEFT patch is not applied successfully. And PEFT has been changed a lot since this came out. The version in requirements.txt is probably no longer correct. Plus this is all windows users installing pre-compiled wheels.
> 
> This is the patch that has to succeed on PEFT: https://github.com/johnsmith0031/alpaca_lora_4bit/blob/main/monkeypatch/peft_tuners_lora_monkey_patch.py
> 
> I have not checked recently what else is going on as it's a good idea to used dedicated training tools to train.

I was able to set the [peft_tuners_lora_monkey_patch.py](https://github.com/johnsmith0031/alpaca_lora_4bit/blob/main/monkeypatch/peft_tuners_lora_monkey_patch.py) in through conda but still getting a lib not found on monkeypatch

**Comment by Ph0rk0z** [comment]
At least the compiling part is. Training.py should be invoking that patch from this code after you loaded the model through the "monkeypatch" itself.

```
    if shared.args.monkey_patch:
        from monkeypatch.peft_tuners_lora_monkey_patch import (
            replace_peft_model_with_gptq_lora_model
        )
        replace_peft_model_with_gptq_lora_model()
```

I have this set up different where I can just check autograd in the UI and it loads. I mainly used it to apply lora and bench vs AutoGPTQ rather than training.

**Comment by webbigdata-jp** [comment]
In my case it was resolved by
```
pip install peft==0.5.0
```

**Comment by Ph0rk0z** [comment]
It's good to know that PEFT still works with the patch.

**Comment by tamil-acog** [comment]
Hi.. I have monkeypatch installed from alpaca_lora_4bit

I still face the issue. 

Trying to train in Linux with 4090(24GB RAM). 

Any help would be appreciated. Thanks

**Comment by kevkid** [comment]
> ing.py should be invoking that patch from this code after you loaded the model through the "monkeypatch" itself.
> 
> ```
>     if shared.args.monkey_patch:
> ```
I am getting the same here

**Comment by Keeo** [comment]
Downgrading the peft to version 0.6.0 solved the issue for me.
 `pip install "peft==0.6.0"`

**Comment by Keeo** [comment]
I was also strugling with this and downgrading verion to 0.5.0 or 0.6.0 did not fix it. What ultimately solved it for me and for now is to uninstall peft and install it from git/source directly.
```
pip uninstall peft
git clone https://github.com/huggingface/peft
cd peft 
pip install -v .
```

**Comment by MehdiGhorb** [comment]
After installing `peft==0.5.0` and also installing `peft` directly from huggingface, I'm still getting the same error.
Has anyone found an alternative way?!
